From 4ba4ee03ae1692eeb6d4552348ae39e1644a1f36 Mon Sep 17 00:00:00 2001
From: Cosimo Cecchi <cosimoc@gnome.org>
Date: Fri, 11 May 2012 15:36:05 +0000
Subject: box: set orientation style classes when setting orientation

Since we override the property, we have to call the internal helper
manually.
---
diff --git a/gtk/gtkbox.c b/gtk/gtkbox.c
index 96bc318..d3a6f78 100644
--- a/gtk/gtkbox.c
+++ b/gtk/gtkbox.c
@@ -81,6 +81,7 @@
 #include "gtkboxprivate.h"
 #include "gtkintl.h"
 #include "gtkorientable.h"
+#include "gtkorientableprivate.h"
 #include "gtkprivate.h"
 #include "gtktypebuiltins.h"
 #include "gtksizerequest.h"
@@ -353,6 +354,7 @@ gtk_box_set_property (GObject      *object,
     {
     case PROP_ORIENTATION:
       private->orientation = g_value_get_enum (value);
+      _gtk_orientable_set_style_classes (GTK_ORIENTABLE (box));
       gtk_widget_queue_resize (GTK_WIDGET (box));
       break;
     case PROP_SPACING:

